examples/vpc-peering-common/main.tf

```diff
@@ -0,0 +1,200 @@
+  vmseries_image        = var.vmseries_common.vmseries_image
+  machine_type          = each.value.machine_type
+  min_cpu_platform      = try(each.value.min_cpu_platform, "Intel Cascade Lake")
+  tags                  = try(each.value.tags, [])
+  service_account       = module.iam_service_account[each.value.service_account].email
+  scopes                = try(each.value.scopes, [])
+  create_instance_group = true
+
+  bootstrap_options = try(merge({ vmseries-bootstrap-gce-storagebucket = "${module.bootstrap[each.value.bootstrap-bucket-key].bucket_name}/${each.key}/" }, var.vmseries_common.bootstrap_options), merge(
+    try(each.value.bootstrap_options, {}),
+    try(var.vmseries_common.bootstrap_options, {})
+  ))
+
+  named_ports = try([for v in each.value.named_ports :
+    {
+      name = v.name
+      port = v.port
+  }], [])
+
+  network_interfaces = [for v in each.value.network_interfaces :
+    {
+      subnetwork       = module.vpc.subnetworks["${var.name_prefix}${v.subnetwork}"].self_link
+      private_ip       = v.private_ip
+      create_public_ip = try(v.create_public_ip, false)
+  }]
+}
+
+data "google_compute_image" "my_image" {
+  family  = "ubuntu-pro-2204-lts"
+  project = "ubuntu-os-pro-cloud"
+}
+
+resource "google_compute_instance" "linux_vm" {
+  for_each = var.linux_vms
+
+  name         = "${var.name_prefix}${each.key}"
+  machine_type = each.value.linux_machine_type
+  zone         = each.value.zone
+
+  boot_disk {
+    initialize_params {
+      image = data.google_compute_image.my_image.id
+      size  = each.value.linux_disk_size
+    }
+  }
+
+  network_interface {
+    subnetwork = module.vpc.subnetworks["${var.name_prefix}${each.value.subnetwork}"].self_link
+    network_ip = each.value.private_ip
+  }
+
+  metadata = {
+    ssh-keys       = each.value.ssh_keys_linux
+    enable-oslogin = true
+  }
+
+
+  service_account {
+    email  = module.iam_service_account[each.value.service_account].email
+    scopes = each.value.scopes
+  }
+}
+
+module "lb_internal" {
+  source = "../../modules/lb_internal"
+
+  for_each = var.lbs_internal
+
+  name              = "${var.name_prefix}${each.value.name}"
+  health_check_port = try(each.value.health_check_port, "80")
+  backends          = { for v in each.value.backends : v => module.vmseries[v].instance_group_self_link }
+  ip_address        = each.value.ip_address
+  subnetwork        = module.vpc.subnetworks["${var.name_prefix}${each.value.subnetwork}"].self_link
+  network           = module.vpc.networks["${var.name_prefix}${each.value.network}"].self_link
+  all_ports         = true
+}
+
+module "lb_external" {
+  source = "../../modules/lb_external"
+
+  project = var.project
+
+  for_each = var.lbs_external
+
+  name                    = "${var.name_prefix}${each.value.name}"
+  backend_instance_groups = { for v in each.value.backends : v => module.vmseries[v].instance_group_self_link }
+  rules                   = { for k, v in each.value.rules : k => v }
```

Comment:
If a user configures a legacy rule, then module invocation should take that into account and also provide `instances` parameter. Both `intences` and `backend_instance_groups` can be specified at the same time without any conflict.

Reply:
Will apply this in a future update.